sale_product_approval/models/product_product.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sale_ok = fields.Boolean(
+        string="Can be added to Quote",
+        copy=False,
+        readonly=True,
+    )
+    sale_ok_confirm = fields.Boolean(
+        string="Can be Sold",
+        copy=False,
+        readonly=True,
+    )
+    candidate_sale = fields.Boolean(
+        string="Candidate to add to Quote",
+    )
+    candidate_sale_confirm = fields.Boolean(
+        string="Candidate to be Sold",
+    )
+    can_edit_candidate = fields.Boolean(compute="_compute_can_edit_candidate")
```

Comment:
This name should maybe reflect 'product approval' flow.

Reply:
We went over the naming of these fields internally quite a bit and ended up settling on 'candidate' because the approval part is in the states and having the 'approved to be' on the products confused the users because they struggled to make the connection between the setting on the product depends on the state. So we settled on 'Candidate' to say basically its configured to be 'Sold', 'Purchased', etc as long as the state it's in allows it.